mode/type/cluster/core/src/main/java/org/apache/shardingsphere/mode/manager/cluster/event/subscriber/dispatch/ResourceMetaDataChangedSubscriber.java

```diff
@@ -128,7 +129,9 @@ public synchronized void renew(final DropViewEvent event) {
     
     private void refreshShardingSphereStatisticsData() {
         PersistRepository repository = contextManager.getPersistServiceFacade().getRepository();
-        if (repository instanceof ClusterPersistRepository) {
+        if (contextManager.getComputeNodeInstanceContext().isCluster()
+                && InstanceType.PROXY == contextManager.getComputeNodeInstanceContext().getInstance().getMetaData().getType()
+                && repository instanceof ClusterPersistRepository) {
```

Comment:
useless instanceof , if contextManager is cluster, the repository must be ClusterPersistRepository